README
This module handles generating name days from dates and vice versa.

INSTALLATION

To install this module type the following in the distribution
directory:

   perl Makefile.PL
   make
   make test
   make install

or use a tool like the CPAN shell or CPAN-Minus (cpanm).


Changes
0.03  2008-xx-xx

- Fix the regexp starting with /^*
- Fix test scripts, so that "my" variables do not hide other "my" variables
- Fix _make_recurrence, to deal with infinite dates

0.02  2003-04-27

- Second release

0.01  2003-04-26

- First release


Makefile.PL
#
#     Configuration script for DateTime::Event::NameDay
#     Copyright (C) 2003, 2018 Ben Bennett and Jean Forget
#
#     This program is distributed under the same terms as Perl 5.28.0:
#     GNU Public License version 1 or later and Perl Artistic License
#
#     You can find the text of the licenses in the F<LICENSE> file or at
#     L<http://www.perlfoundation.org/artistic_license_1_0>
#     and L<https://www.gnu.org/licenses/gpl-1.0.html>.
#
#     Here is the summary of GPL:
#
#     This program is free software; you can redistribute it and/or modify
#     it under the terms of the GNU General Public License as published by
#     the Free Software Foundation; either version 1, or (at your option)
#     any later version.
#
#     This program is distributed in the hope that it will be useful,
#     but WITHOUT ANY WARRANTY; without even the implied warranty of
#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#     GNU General Public License for more details.
#
#     You should have received a copy of the GNU General Public License
#     along with this program; if not, write to the Free Software Foundation,
#     Inc., <https://www.fsf.org/>.
#

use ExtUtils::MakeMaker;
WriteMakefile(
    NAME         => 'DateTime::Event::NameDay',
    VERSION_FROM => 'lib/DateTime/Event/NameDay.pm',
    PREREQ_PM    => { DateTime 	                    => 0,
                      DateTime::Set                 => 0,
                      DateTime::Calendar::Christian => 0,
                      Test::More                    => 0,
                      Carp                          => 0,
                      Params::Validate              => 0,
                     'vars'                         => 0,
                     'strict'                       => 0,
                     'warnings'                     => 0,
                       },
    AUTHOR       => [ 'Ben Bennett (fiji@limey.net)', 'Jean Forget <JFORGET@cpan.org>' ],
    ABSTRACT     => 'Namedays from various countries',
    LICENSE        => 'perl',
    BUILD_REQUIRES => { 'Test::More'         => '0',        # should be 'TEST_REQUIRES', except that ExtUtils::MakeMaker 6.5705 does not accept it
                        'ExtUtils::MakeMaker' => '6.57_02', # the first version to accept several authors in an arrayref
                           },
    MIN_PERL_VERSION => '5.8.8', # IIRC, the first version with a good support of Unicode
    META_MERGE       => {
       dynamic_config => 0,
       provides       => {
           'DateTime::Event::NameDay' => {
                file => 'lib/DateTime/Event/NameDay.pm',
                version => '0.02',
           },
           'DateTime::Event::NameDay::france' => {
                file => 'lib/DateTime/Event/NameDay/france.pm',
                version => '0.02',
           },
           'DateTime::Event::NameDay::greece' => {
                file => 'lib/DateTime/Event/NameDay/greece.pm',
                version => '0.02',
           },
           'DateTime::Event::NameDay::slovakia' => {
                file => 'lib/DateTime/Event/NameDay/slovakia.pm',
                version => '0.02',
           },
           'DateTime::Event::NameDay::sweden' => {
                file => 'lib/DateTime/Event/NameDay/sweden.pm',
                version => '0.02',
           },
       },
       'meta-spec' => { version => 2 },
       resources => {
           repository => {
               type => 'git',
               url  => 'https://github.com/jforget/DateTime-Event-NameDay.git',
               web  => 'https://github.com/jforget/DateTime-Event-NameDay',
           },
       },
       prereqs => {
         build => {
           requires => {
             'strict'                         => 0,
             'warnings'                       => 0,
           },
         },
         runtime => {
           requires => {
             'vars'                           => 0,
             'strict'                         => 0,
             'warnings'                       => 0,
             'Carp'                           => 0,
             'DateTime'                       => 0,
             'DateTime::Set'                  => 0,
             'DateTime::Calendar::Christian'  => 0,
             'Params::Validate'               => 0,
             'perl'                           => '5.6.1',
           },
         },
         test => {
           requires => {
             'Test::More'                     => '0',
             'strict'                         => 0,
             'warnings'                       => 0,
           },
         },
       },
    },
);


lib/DateTime/Event/NameDay.pm
#
#     Perl extension to work with namedays from various countries.
#     Copyright (C) 2003, 2018 Ben Bennett and Jean Forget
#
#     See the license in the embedded documentation below.
#
package DateTime::Event::NameDay;

use utf8;
use strict;
use warnings;

use vars qw ($VERSION);

$VERSION = '0.02';

use Carp;
use Params::Validate qw( validate SCALAR OBJECT HASHREF );

use DateTime;
use DateTime::Set;
use DateTime::Calendar::Christian;


my %namedays = ();


sub new {
    my $class = shift;
    my %p = validate( @_,
		      { country   => { type      => SCALAR,
				       default   => undef,
				       # Leave the heavy validation to set
				     },
			date_args => { type      => HASHREF,
				       default   => {},
				     },
		       },
		      );
    my $self = { };
    bless $self, $class;
    $self->set( %p );

    return $self;
}

sub set {
    my $self = shift;

    my %p = validate( @_,
                      { country   => { type      => SCALAR,
				       optional  => 1,
				       callbacks => 
				           {'known day mapping' => \&_check_country }
				     },
			date_args => { type      => HASHREF,
				       default   => {},
				     },
		      }
		      );

    if (defined $p{country}) {
	$self->{country} = lc $p{country};
    }

    if (defined $p{date_args}) {
	$self->{date_args} = $p{date_args};
    }

    return $self;
}

sub country {
    my ($self) = @_;
    return undef unless ref $self;
    return $self->{country};
}


sub date_args {
    my ($self) = @_;
    return {} unless ref $self;
    return $self->{date_args};
}

sub get_daynames
{
    my $self = shift;
    my %p = validate( @_,
                      { country => { type      => SCALAR,
				     optional  => 1,
				     callbacks => 
				           {'known day mapping' => \&_check_country }
				     },
			date    => { type      => OBJECT,
				     can       => 'utc_rd_values',
				     },
		      }
		      );

    # Work out our country
    my $country = lc $p{country};
    if (not defined $country) {
	$country = $self->country();

	croak "Unable to determine the correct country"
	    unless defined $country;
    }

    # Get the namedays for the given date
    # - Find our section
    my $nameday_info = 
	$self->_init_nameday_country(namedays => \%namedays,
				     country  => $country);

    # - Convert to the Julian calendar
    my $adj_dt = DateTime::Calendar::Christian->from_object
	(object      => $p{date},
	 reform_date => $nameday_info->{reform_date},
	 %{ $self->date_args() },
	 );
    
    # - Get the appropriate nameday based on month number and day
    my $names = $nameday_info->{names}{ $adj_dt->month() }{ $adj_dt->day() };
    my @names = defined $names ? @$names : ();

    return @names;
}

sub get_namedays {
    my $self = shift;
    my %p = validate( @_,
                      { country => { type      => SCALAR,
				     optional  => 1,
				     callbacks => 
				           {'known day mapping' => \&_check_country }
				     },
			date_args => { type      => HASHREF,
				       default   => undef,
				     },
			name    => { type      => SCALAR,
				     },
		      }
		      );

    # Work out our country
    my $country = lc $p{country};
    if (not defined $country) {
	$country = $self->country();

	croak "Unable to determine the correct country"
	    unless defined $country;
    }

    # Work out the date args
    my $date_args = $p{date_args};
    if (not defined $date_args) {
	$date_args = ref $self ? $self->date_args() : {};
    }

    # Get the canonical name
    my $name = _clean_name( $p{name} );

    # Find the month and day for the given name
    my $nameday_info = 
	$self->_init_nameday_country(namedays => \%namedays,
				     country  => $country);
    croak "Unknown name '$p{name}' for country '$p{country}'"
	unless exists $nameday_info->{reverse_names}{$name};
    my ($month, $day) = @{ $nameday_info->{reverse_names}{$name} };

    # Build a set of all of the days that the given name is for
    my $set = DateTime::Set->from_recurrence
	(next => 
	     sub { _make_recurrence($_[0], $nameday_info->{reform_date},
					   $month, $day, 1, $date_args);
	     },
	 previous => 
	     sub { _make_recurrence($_[0], $nameday_info->{reform_date},
					   $month, $day, -1, $date_args);
	     },
	 );

    return $set;
}

sub _make_recurrence {
    my ($last, $reform_date, $month, $day, $direction, $date_args) = @_;

    return $last if $last->is_infinite;

    my $dt = DateTime::Calendar::Christian->from_object
	(object      => $last,
	 reform_date => $reform_date,
	 %$date_args,
	 );
    $dt->truncate(to => 'day');
    my $target = $dt->clone();
    $target->set( month => $month,
		  day   => $day,
		  );
    
    if ($direction == 1) {
	if ($dt >= $target) {
	    $target->add( years => 1);
	}
    } else {
	if ($dt <= $target) {
	    $target->subtract( years => 1);
	}
    }

    $target->set( month => $month,
		  day   => $day,
		  );

    return DateTime->from_object(object => $target, %$date_args);
}

sub _init_nameday_country {
    my $self = shift;
    my %args = @_;
    my $country  = $args{country};
    my $namedays = $args{namedays};


    return $namedays->{$country}
        if exists $namedays->{$country};
    
    # Okay, load the nameday info from the sub item
    my $package = "DateTime::Event::NameDay::$country";
    eval "require $package;";
    
    my ($reform_date, $data) = $package->nameday_data();
    my @data = split /\n/, $data;
    undef $data;

    $namedays->{$country} = 
        {reform_date   => $reform_date,
	 names         => {},
	 reverse_names => {},
	 };
    my $forward = $namedays->{$country}{names};
    my $reverse = $namedays->{$country}{reverse_names};
    

    # Format of files:
    # - #s are comments, blank lines are ignored
    # - leading whitespace and trailing whitespace is ignorred
    # - months occur by number on their own line followed by a : (i.e. 1:)
    # - name days are given by number followed by a space then a , separated list of names
    #   surrounding whitespace is trimmed
    # - names starting with * are immovable holidays
    # 
    # e.g.:
    # # Nameday file
    # # Source: www.whatever.com
    #
    # 1: # January
    #   1 *New Year's Day
    #   2 Svea
    #   3 Alfred, Alfrida
    #   ...
    #
    # 2: # February
    #   ...

    my $month = undef;
    foreach my $line (@data) {
	$line =~ s/\s*\#.*//;
	next if $line =~ /^\s*$/;
	
	if ($line =~ /^\s*(\d+)\s+(.*)/) {
	    my ($day, $names) = ($1, $2);

	    # We have a day
	    croak "Malformed nameday file for '$package': Missing month before line '$line'"
		unless defined $month;

	    # Split the names apart and store the forward mapping
	    # For the moment remove the * indicating holidays
	    $names =~ s/\s+$//;
	    my @names = map { s/^\*//; $_ } split /\s*,\s*/, $names;
	    $forward->{$month}{$day} = \@names;

	    # Store the reverse mapping
	    foreach my $name (@names) {
		my $n = _clean_name($name);
		croak "Duplicate name '$name' (cleaned '$n')"
		    if exists $reverse->{$n};
		$reverse->{$n} = [$month, $day];
	    }
	}
	elsif ($line =~ /^\s*(\d+):\s*$/) {
	    # Change the month
	    $month = $1;
	}
    }
    @data = ();

    return $namedays->{$country};
}

# Get the canonical sort string for the name... all lowercase
# I would like to make it insensitive to accents too, but there 
# is no good way to do that yet
sub _clean_name {
    my $name = shift;

    $name = lc $name;
#    $name = NFD( lc($name) );
#    $name =~ s/\pM//g;

    return $name;
}

# See if the given country is one we support
sub _check_country {
    my $country = lc shift;
    
    # See if we can load the module
    my $package = "DateTime::Event::NameDay::$country";
    eval "require $package;";

    return $@ ? 0 : 1;
}

# Ending a module with "1" is so boring. Using some other true value is fun
'Par sainte Barbe, vive la sape !';

__END__

=head1 NAME

DateTime::Event::NameDay - Perl DateTime extension to work with namedays from various countries.

=head1 SYNOPSIS

  use DateTime::Event::NameDay;

  $nd = DateTime::Event::NameDay->new
    (country => "Sweden" , 
     date_args => {reform_date => $ref_dt} );

  @names = $nd->get_daynames( date => $dt);
  # Returns an array of the matching names

  $set = $nd->get_namedays(name => "Alfred");
  # Returns a DateTime::Set object that represents all of the 
  # namedays for Alfred

  # Alternately you can call it without using an object, but
  # now you must specify the country rather than using the
  # one (optionally) passed to the constructor
  @names = DateTime::Event::NameDay->get_daynames
               ( country => "France", date => $dt);
  $set   = DateTime::Event::NameDay->get_namedays
               ( country => "France", name => "Basile");


=head1 DESCRIPTION

DateTime::Event::NameDay is a class that knows the name days for
various countries.  In some countries a person's nameday is more
important than their birthday and gifts may be exchanged.

There are two major functions of the class, the first takes a
C<DateTime> object (of any kind) and works out the names that are
associated with that day.  The second takes a given name and returns a
C<DateTime::Set> object that can be used to work out what dates are
for the given name.

Please note that the calculations are done using the
C<DateTime::Calendar::Christian> module to deal with dates that fall
before the calendar reforms.

=head1 USAGE

TODO

=head1 AUTHOR

First author: Ben Bennett <fiji at limey dot net>

Unofficial co-maintainer: Jean Forget <JFORGET@cpan.org>

=head1 COPYRIGHT

Copyright  (c) 2003,  2018 Ben  Bennett  and Jean  Forget. All  rights
reserved.  This program  is  free software;  you  can redistribute  it
and/or modify it under the same terms as Perl itself.

Portions of the code in this distribution are derived from other
works.  Please see the CREDITS file for more details.

You can find the text of the licenses in the F<LICENSE> file with this
module or at L<https://dev.perl.org/licenses/artistic.html> and
L<https://www.gnu.org/licenses/gpl-1.0.html>.

Here is the summary of GPL:

This program is  free software; you can redistribute  it and/or modify
it under the  terms of the GNU General Public  License as published by
the Free  Software Foundation; either  version 1, or (at  your option)
any later version.

This program  is distributed in the  hope that it will  be useful, but
WITHOUT   ANY  WARRANTY;   without  even   the  implied   warranty  of
MERCHANTABILITY  or FITNESS  FOR A  PARTICULAR PURPOSE.   See  the GNU
General Public License for more details.

You should  have received  a copy  of the  GNU General  Public License
along with  this program; if not,  see <https://www.gnu.org/licenses/>
or write to the Free Software Foundation, Inc., L<https://fsf.org>.

=head1 SEE ALSO

datetime@perl.org mailing list

L<http://datetime.perl.org/>

=cut


lib/DateTime/Event/NameDay/greece.pm
#
#     Perl extension to work with namedays from Greece
#     Copyright (C) 2003, 2018 Ben Bennett and Jean Forget
#
#     See the license in the embedded documentation below.
#
package DateTime::Event::NameDay::greece;

use utf8;
use strict;
use warnings;
use DateTime;

sub nameday_data {
    
    return(DateTime->new( year => 1923, month => 2, day => 28),
           <<"EOF;");
# From:       http://www.thegreekforum.com/nameandholydays.php
# Country:    Greece
# ReformDate: 1923/02/28 (a decent guess, different parts reformed at different times)

1: # January
  1 Vasilis
  6 Fotine, Fotini, Fotis, Theofania, Theofanis
  7 Ioanna, Ioannis, Prodromos, Yanna, Yannis
  8 Theofilos
 11 Theodosios, Theodosis
 12 Tatiane, Tatiani
 17 Antonia, Antonios, Antonis
 18 Athanasios, Kyrillos, Thanasis
 19 Makarios
 20 Evthymios, Thymios
 21 Agne, Agni, Maximos, Neofytos
 22 Timotheos
 23 Agathangelos
 24 Polyxene, Polyxeni, Xene, Xeni, Zosimas
 25 Gregorios, Gregoris, Grigorios, Grigoris, Margarita
 26 Xenofon
 27 Chrysostomos
 28 Palladios
 31 Evdoxia

2: # February
  1 Tryfonas
  2 Ypapante, Ypapanti
  4 Isidoros
  5 Agathe, Agathi
  6 Fotis
  7 Parthenios, Parthenis
  8 Zacharias
 10 Chara, Charalambos, Chariklia
 11 Vlassis
 12 Meletios, Meletis
 14 Valentina, Valentine, Valentini, Valentinos
 15 Evsevios
 19 Filothei
 23 Polykarpos
 26 Fotine, Fotini, Porfyrios
 27 Asklepios, Asklipios

3: # March
  1 Evdokia
  2 Evthalia
 12 Theofania, Theofanis
 15 Theodoros *
 16 Christodoulos
 17 Alekos, Alexios, Alexis
 19 Chrysanthos
 21 Iakovos
 25 Evangelos, Vangelis

4: # April
  6 Evtychios
  9 Iosif, Sifis
 10 Herakles
 12 Anthe, Anthi
 15 Leonidas
 19 Lazaros *
 20 Vaia *, Vaios *
 22 Nathanael
 23 Georgios, Giorgos, Yorgos
 24 Elisavet
 25 Markela, Markos, Nike, Niki
 27 Anastasios *, Lambros *, Paschalis *
 29 Iason

5: # May
  2 Pege *, Pigi *, Zoe *, Zoi *
  4 Pelagia, Thomas *
  5 Eirene, Eirini, Irene, Irini
  6 Serafeim
  8 Arsenios, Theologos
  9 Christoforos, Isaias
 10 Simonas
 11 Methodios
 13 Glykeria, Sergios
 14 Isidoros
 15 Achilleas
 17 Andronikos, Iounia
 18 Ioulia
 20 Lydia
 21 Elena, Elene, Eleni, Konstantina, Konstantinos, Lena
 29 Theodosia

6: # June
  2 Nikiforos
  4 Martha
  5 Apollon, Dorotheos
  8 Kalliope, Kalliopi
 11 Varnavas, Vartholomaios
 16 Tychon
 19 Paisios
 25 Fevronia
 29 Pavlos, Petros
 30 Apostolis, Apostolos

7: # July
  1 Anargyros, Argyris, Damianos, Kosmas
  4 Loukia
  7 Kyriake, Kyriaki
  8 Prokopios, Theofilos
 11 Evfemia, Evfimia, Olga
 12 Veronike, Veroniki
 14 Akylas, Nikodemos, Nikodimos
 15 Ioulitta, Kerykos, Kirykos
 17 Marina
 18 Aimilianos
 20 Elias, Ilias
 22 Magdalena, Magdalene, Magdalini, Markella
 24 Christina
 25 Anna
 26 Erse, Ersi, Paraskeve, Paraskevi
 27 Pantelis
 29 Kallinikos
 30 Andronikos
 31 Iosif, Sifis

8: # August
  6 Sotiria, Sotiris
  7 Astero
  8 Triantafyllos
 15 Despoina, Maria, Panagiotis, Panayotis
 16 Diomedes, Diomidis
 18 Flora, Floros
 20 Samouel
 25 Titos
 26 Adrianos, Natalia
 27 Fanourios
 30 Alexandra, Alexandros

9: # September
  1 Antigone, Antigoni, Athena, Athina, Margarita, Polynike, Polyniki, Symeon
  3 Anthimos, Phoebe, Phoebi
  4 Ermione, Ermioni
  5 Zacharias
  9 Ioakeim
 11 Evanthia
 13 Aristeides, Aristeidis, Kornelios
 14 Stavros, Stavroula
 15 Nikitas
 16 Evfemia
 17 Agape, Agapi, Elpida, Sofia, Sophia
 18 Ariadne, Ariadni
 20 Evstathios, Stathis
 22 Fokas
 23 Xanthippe, Xanthippi
 24 Thekla
 25 Evfrosyne, Evfrosyni
 27 Zenon, Zinon
 29 Kyriakos

10: # October
  2 Kyprianos
  3 Dionysis
  4 Ierotheos
  5 Charitine, Charitini
  7 Polychronios, Sergios
  8 Pelagia
  9 Avraam
 10 Evlambios
 15 Loukianos
 18 Loukas
 20 Artemios, Gerasimos, Yerasimos
 21 Sokrates, Sokratis
 23 Iakovos
 26 Demetris, Dimitra, Dimitrios, Dimitris
 27 Nestor
 30 Zenovia, Zenovios, Zinovia, Zinovios

11: # November
  1 Anargyros, Argyris, Damianos, Kosmas
  8 Angela, Angelos, Gavriel, Michalis, Stamatis, Stamos, Taxiarchis
  9 Nektarios
 11 Minas, Viktoras
 13 Chrysostomos
 14 Filippos
 16 Mathaios, Matthaios
 18 Platonas
 21 Maria
 22 Filemon, Filimon
 25 Aikaterine, Aikaterini, Katerina, Merkourios
 26 Stelios, Stergios, Stylianos
 30 Andreas, Andriane, Andriani

12: # December
  4 Varvara
  5 Diogenes, Savas, Savvas
  6 Nikolaos, Nikoleta, Nikos
  7 Amvrosios
  9 Anna
 12 Spiros, Spyridon, Spyros
 13 Evstratios, Loukia, Stratos
 15 Elevtherios, Lefteris
 17 Daniel, Dionysis
 19 Aglaia
 20 Ignatios
 21 Themistokles, Themistoklis
 22 Anastasia
 24 Evgenia
 25 Christina, Christos, Chrysa
 26 Emmanouil, Manolis
 27 Stefanos, Stephanos
EOF;
}

1;

__END__

=head1 DESCRIPTION

This module cannot be used alone. It must be used with
C<DateTime::Event::NameDay>. Please refer to this module.

=head1 AUTHOR

First author: Ben Bennett <fiji at limey dot net>

Unofficial co-maintainer: Jean Forget <JFORGET@cpan.org>

=head1 COPYRIGHT

Copyright  (c) 2003,  2018 Ben  Bennett  and Jean  Forget. All  rights
reserved.  This program  is  free software;  you  can redistribute  it
and/or modify it under the same terms as Perl itself.

Portions of the code in this distribution are derived from other
works.  Please see the CREDITS file for more details.

You can find the text of the licenses in the F<LICENSE> file with this
module or at L<https://dev.perl.org/licenses/artistic.html> and
L<https://www.gnu.org/licenses/gpl-1.0.html>.

Here is the summary of GPL:

This program is  free software; you can redistribute  it and/or modify
it under the  terms of the GNU General Public  License as published by
the Free  Software Foundation; either  version 1, or (at  your option)
any later version.

This program  is distributed in the  hope that it will  be useful, but
WITHOUT   ANY  WARRANTY;   without  even   the  implied   warranty  of
MERCHANTABILITY  or FITNESS  FOR A  PARTICULAR PURPOSE.   See  the GNU
General Public License for more details.

You should  have received  a copy  of the  GNU General  Public License
along with  this program; if not,  see <https://www.gnu.org/licenses/>
or write to the Free Software Foundation, Inc., L<https://fsf.org>.



lib/DateTime/Event/NameDay/france.pm
#
#     Perl extension to work with namedays from France
#     Copyright (C) 2003, 2018 Ben Bennett and Jean Forget
#
#     See the license in the embedded documentation below.
#
package DateTime::Event::NameDay::france;

use utf8;
use strict;
use warnings;
use DateTime;

sub nameday_data {
    
    return("france",
           <<"EOF;");
# From:       http://meilleursprenoms.com/site/Fetes/Fetes.htm
# Country:    France
# ReformDate: 1582-12-20

1: # January
  1 *Jour de l'an #'
  2 Basile
  3 Geneviève
  4 Odilon
  5 Edouard
  6 Balthazar, Mélaine, Melchior, Tiffany
  7 Aldric, Cédric, Raymond
  8 Lucien
  9 Alix
 10 Guillaume
 11 Hortense, Pauline
 12 Tatiana
 13 Hilaire, Yvette
 14 Nina
 15 Rachel, Rémi
 16 Marcel
 17 Roseline
 18 Gwendal, Prisca
 19 Marius
 20 Fabien, Sébastien
 21 Agnès
 22 Vincent
 23 Banard
 24 François
 26 Timothé # Pauline
 27 Angèle
 28 Manfred, *Thomas d'Aquin #'
 29 Gildas
 30 Jacinthe, Martine
 31 Marcelle

2: # February
  1 Ella, Siméon
  2 Théophane
  3 Blaise, Nelson, Oscar
  4 Véronique
  5 Agathe
  6 Dorothée, Gaston
  7 Eugénie
  8 Jacqueline
  9 Apolline
 10 Arnaud
 11 Lourdes, *Notre Dame de Lourdes
 12 Félix
 13 Béatrice
 14 Valentin
 15 Claude, Georgina, Jordan
 16 Julienne, Lucile, Onésime
 17 Alexis
 18 Bernadette
 19 Gabin
 20 Aimée
 21 Damien
 22 Isabelle
 23 Lazare
 24 Modeste
 25 Roméo
 26 Nestor
 27 Honorine, Léandre
 28 Romain
 29 Auguste

3: # March
  1 Albin, Aubin, Jonathan
  2 *Charles le Bon
  3 Guénolé, Marin
  4 Casimir
  5 Olivia
  6 Colette
  7 Félicie, Nathan
  8 *Jean de Dieu
  9 Françoise
 10 Vivien
 11 Rosine
 12 Justine, Pol
 13 Rodrigue
 14 Mathilde
 15 Louise
 16 Bénédicte
 17 Patrice, Patrick
 18 Cyrille
 19 Joseph
 20 *Printemps
 21 Axelle, Clémence
 22 Léa
 23 Rébecca, Victorien
 24 Karine # Catherine
 25 Humbert
 26 Larissa
 27 Habib
 28 Gontran
 29 Gladys
 30 Amédée
 31 Benjamin

4: # April
  1 Hugues, Valéry
  2 Sandrine
  3 Richard
  4 Isidore
  5 Irène
  6 Marcellin
  7 Clotaire, *Jean-Baptiste de la Salle
  8 Julie
  9 Gautier
 10 Fulbert
 11 Stanislas
 12 Jules
 13 Ida
 14 Ludivine, Maxime
 15 César
 16 Benoît-Joseph # Bernadette
 17 Anicet
 18 Parfait
 19 Emma
 20 Odette, Théotime
 21 Anselme
 22 Alexandre
 23 Georges
 24 Fidèle
 25 Marc
 26 Alida
 27 Zita
 28 Valérie
 29 *Catherine de Sienne
 30 Robert

5: # May
  1 Brieuc, Florine, Jérémie, Tamara
  2 Boris, Zoé
  3 Ewen, Philippe # Jacques
  4 Florian, Sylvain
  5 Judith
  6 Marien, Prudence
  7 Domitille, Gisèle
  8 Désiré
  9 Pacôme
 10 Solange
 11 Estelle, Mayeul
 12 Achille
 13 Maël, Orlane, Rolande
 14 Aglaé, Matthias
 15 Denise
 16 Brendan, Honoré
 17 Pascal
 18 Eric
 19 Célestin, Erwan, Yves
 20 Bernardin
 21 Constantin
 22 Emile, Quitterie, Rita
 23 Didier
 24 Donatien
 25 Sophie
 26 Bérenger
 27 Augustin
 28 Germain
 29 Aymar, Géraldine, Maximin
 30 Ferdinand, Jeanne, Lorraine
 31 Pétronille

6: # June
  1 Justin, Ronan
  2 Blandine
  3 Kévin
  4 Clotilde
  5 Igor
  6 Norbert
  7 Gilbert
  8 Médard
  9 Diane
 10 Landry
 11 Barnabé
 12 Guy
 13 *Antoine de Padoue
 14 Elisée, Valère
 15 Germaine
 16 François-Régis, Régis
 17 Hervé
 18 Léonce
 19 Gervais, Romuald
 20 Silvère
 21 Eté
 22 Alban
 23 Audrey
 24 Jean-Baptiste
 25 Aliénor, Eléonore, Prosper, Salomon
 26 Anthelme
 27 Fernand
 28 Irénée
 29 Paul, Pierre
 30 Adolphe, Martial

7: # July
  1 Aaron, Esther, Goulwen, Thierry
  2 Martinien
  3 Thomas
  4 Florent
  5 Antoine
  6 Mariette, Nolwen
  7 Raoul
  8 Edgar, Killian, Priscillia, Thibault
  9 Amandine, Hermine, Iphigénie, Marianne
 10 Ulrich
 11 Benoit, Olga, Yolande
 12 Jason, Olivier
 13 Enzo, Eugène, Henri, Joël
 14 Camille
 15 Donald, Vladimir
 16 Elvire
 17 Arlette, Charlotte, Marcelline
 18 Frédéric
 19 Arsène, Micheline
 20 Elie, Marina
 21 Rodolphe, Térence, Victor
 22 Madeleine, Wandrille
 23 Brigitte
 24 Christine, Ségolène
 25 Jacques, Valentine
 26 Anne, Hannah, Joachin
 27 Aurèle, Nathalie
 28 Samson
 29 Beatrix, Loup, Marthe
 30 Juliette
 31 Ignace, *Ignace de Loyola

8: # August
  1 Alphonse
  2 Julien, *Julien Eymard
  3 Lydie
  4 Vianney
  5 Abel
  6 *Transfiguration
  7 Gaétan
  8 Dominique
  9 Amour
 10 Laurent
 11 Claire, Gilberte, Suzanne
 12 Clarisse
 13 Hippolyte
 14 Evrard
 15 Alfred, Marie
 16 Armel, Roch
 17 Hyacinthe
 18 Hélène, Laétitia
 19 *Jean Eudes
 20 Bernard, Samuel
 21 Christophe, Grâce, Ombeline
 22 Fabrice
 23 Rose, *Rose de Lima
 24 Barthélémy
 25 Louis
 26 Natacha
 27 Monique
 28 Elouan # Augustin
 29 Médéric, Sabine
 30 Fiacre
 31 Aristide

9: # September
  1 Gilles, Jossué
  2 Ingrid
  3 Grégoire
  4 Iris, Moïse, Rosalie
  5 Raïssa
  6 Bertrand, Eva
  7 Reine
  8 Adrien, Béline, Nativité
  9 Alain, Omer
 10 Inès
 11 Adelphe, Glenn, Vinciane
 12 Apollinaire
 13 Aimé
 14 *Croix Glorieuse
 15 Dolores, Roland
 16 Edith
 17 Hildegarde, Lambert, Renaud
 18 Nadège, Véra
 19 Emilie
 20 Davy
 21 Déborah, Jonas, Matthieu, Mélissa
 22 Maurice
 23 Faustine
 24 Thècle
 25 Hermann
 26 Côme # Damien
 27 *Vincent de Paul
 28 Venceslas
 29 Gabriel, Michel, Raphaël
 30 Jérôme

10: # October
  1 Ariel, Mélodie, Muriel, *Thérèse de l'Enfant Jésus #'
  2 Léger, Ruth
  3 Gérard, Sybille
  4 Aure, Bérénice, Frank, Orianne, Sarah, *François d'Assise #'
  5 Camélia, Capucine, Daphne, Eglantin, Fleur, Placide
  6 Bruno
  7 Gustave, Serge
  8 Pélagie, Thaïs
  9 Denis
 10 Ghislain, Virgile
 11 Firmin
 12 Edwin, Séraphin, Wilfried
 13 Géraud
 14 Céleste, Gwendoline, Juste
 15 Thérèse, *Thérèse d'Avila #'
 16 Edwige
 17 Baudoin, Solène
 18 Luc
 19 Cléo, René
 20 Adeline, Aline
 21 Céline, Ursule
 22 Elodie, Salomé, Sara
 23 Simon, *Jean de Capistran
 24 Florentin
 25 Crépin
 26 Dimitri
 27 Emeline
 28 Jude
 29 Narcisse
 30 Bienvenue, Maéva
 31 Quentin

11: # November
  1 Toussaint
  2 Défunts
  3 Gwenaël, Hubert
  4 Aymeric, Charles, Jessé
  5 Sylvie, Zacharie
  6 Bertille, Léonard
  7 Carine
  8 Dora, Geoffroy
  9 Maturin, Théodore
 10 Léon, Noé
 11 Martin, Vérane
 12 Christian
 13 Brice
 14 Sidoine
 15 Albert, Arthur, Léopold, Malo, Victoire
 16 Gertrude, Marguerite, Mégane
 17 Elisabeth, Elise, Hilda
 18 Aude
 19 Tanguy
 20 Edmond, Octave
 21 *Présence de Marie
 22 Cécile
 23 Clément
 24 Flora
 25 Catherine
 26 Delphine
 27 Séverin
 28 *Jacques de la Marche
 29 Saturnin
 30 André, Tugdual

12: # December
  1 Florence
  2 Viviane
  3 Xavier
  4 Barbe, Barbara
  5 Gérald # Gérard
  6 Nicolas
  7 Ambroise
  8 *Immaculée Conception
  9 *Pierre Fourier
 10 Eulaire, Romaric
 11 Daniel
 12 Chantal
 13 Jocelyn, Lucie
 14 Odile
 15 Ninon
 16 Alice
 17 Adélaïde, Gaël, Judicaël, Olympe
 18 Briac, Gatien
 19 Urbain
 20 Isaac, Jacob, Théophile
 21 * Pierre Canis
 22 Françoise-Xavière, Gratien
 23 Armand
 24 Adèle
 25 Manuel
 26 Etienne
 27 Fabiola, Jean
 28 Gaspard
 29 David
 30 Roger
 31 Colombe, Sylvestre
EOF;
}

# Instead of ending the module with a boring "1" value, celebrate the
# saint absent from the calendar
'Saint Glinglin';

__END__

=head1 DESCRIPTION

This module cannot be used alone. It must be used with
C<DateTime::Event::NameDay>. Please refer to this module.

=head1 AUTHOR

First author: Ben Bennett <fiji at limey dot net>

Unofficial co-maintainer: Jean Forget <JFORGET@cpan.org>

=head1 COPYRIGHT

Copyright  (c) 2003,  2018 Ben  Bennett  and Jean  Forget. All  rights
reserved.  This program  is  free software;  you  can redistribute  it
and/or modify it under the same terms as Perl itself.

Portions of the code in this distribution are derived from other
works.  Please see the CREDITS file for more details.

You can find the text of the licenses in the F<LICENSE> file with this
module or at L<https://dev.perl.org/licenses/artistic.html> and
L<https://www.gnu.org/licenses/gpl-1.0.html>.

Here is the summary of GPL:

This program is  free software; you can redistribute  it and/or modify
it under the  terms of the GNU General Public  License as published by
the Free  Software Foundation; either  version 1, or (at  your option)
any later version.

This program  is distributed in the  hope that it will  be useful, but
WITHOUT   ANY  WARRANTY;   without  even   the  implied   warranty  of
MERCHANTABILITY  or FITNESS  FOR A  PARTICULAR PURPOSE.   See  the GNU
General Public License for more details.

You should  have received  a copy  of the  GNU General  Public License
along with  this program; if not,  see <https://www.gnu.org/licenses/>
or write to the Free Software Foundation, Inc., L<https://fsf.org>.

